examples/gnn_explainer_link_pred.py

```diff
@@ -0,0 +1,130 @@
+import os.path as osp
+
+import torch
+from sklearn.metrics import roc_auc_score
+
+import torch_geometric.transforms as T
+from torch_geometric.datasets import Planetoid
+from torch_geometric.explain import Explainer, ExplainerConfig, ModelConfig
+from torch_geometric.explain.algorithm import GNNExplainer
+from torch_geometric.nn import GCNConv
+from torch_geometric.utils import negative_sampling
+
+dataset = 'Cora'
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', 'Planetoid')
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+transform = T.Compose([
+    T.NormalizeFeatures(),
+    T.ToDevice(device),
+    T.RandomLinkSplit(num_val=0.05, num_test=0.1, is_undirected=True,
+                      add_negative_train_samples=False),
+])
+dataset = Planetoid(path, dataset, transform=transform)
+# After applying the `RandomLinkSplit` transform, the data is transformed from
+# a data object to a list of tuples (train_data, val_data, test_data), with
+# each element representing the corresponding split.
+train_data, val_data, test_data = dataset[0]
+
+
+class Net(torch.nn.Module):
+    def __init__(self, in_channels, hidden_channels, out_channels):
+        super().__init__()
+        self.conv1 = GCNConv(in_channels, hidden_channels)
+        self.conv2 = GCNConv(hidden_channels, out_channels)
+
+    def encode(self, x, edge_index):
+        x = self.conv1(x, edge_index).relu()
+        return self.conv2(x, edge_index)
+
+    def decode(self, z, edge_label_index):
+        src, dst = edge_label_index
+        return (z[src] * z[dst]).sum(dim=-1)
+
+    def forward(self, x, edge_index, edge_label_index,
+                cast_to_multiclass=False):
+        z = model.encode(x, edge_index)
+        out = model.decode(z, edge_label_index).view(-1)
+        if cast_to_multiclass:
+            out = out.sigmoid()
+            out = torch.stack((1 - out, out), dim=-1)
+        return out
+
+
+model = Net(dataset.num_features, 128, 64).to(device)
+optimizer = torch.optim.Adam(params=model.parameters(), lr=0.01)
+criterion = torch.nn.BCEWithLogitsLoss()
```

Comment:
In previous examples, we use more `torch.nn.functional`. Therefore I would suggest using:

```
loss = F.binary_cross_entropy_with_logits(out, edge_label)
```

Reply:
done